go-controller/pkg/ovn/ovn_test.go

```diff
@@ -91,32 +91,39 @@ type FakeOVN struct {
 	eIPController  *EgressIPController
 	portCache      *PortCache
 
+	fexec *ovntest.FakeExec
 	// information map of all secondary network controllers
 	secondaryControllers       map[string]secondaryControllerInfo
 	fullSecondaryL2Controllers map[string]*SecondaryLayer2NetworkController
 }
 
-// NOTE: the FakeAddressSetFactory is no longer needed and should no longer be used. starting to phase out FakeAddressSetFactory
 func NewFakeOVN(useFakeAddressSet bool) *FakeOVN {
+	return NewFakeOVNWithExec(useFakeAddressSet, nil)
+}
+
+// NOTE: the FakeAddressSetFactory is no longer needed and should no longer be used. starting to phase out FakeAddressSetFactory
+func NewFakeOVNWithExec(useFakeAddressSet bool, fexec *ovntest.FakeExec) *FakeOVN {
 	var asf *addressset.FakeAddressSetFactory
 	if useFakeAddressSet {
 		asf = addressset.NewFakeAddressSetFactory(DefaultNetworkControllerName)
 	}
 	return &FakeOVN{
-		asf:          asf,
-		fakeRecorder: record.NewFakeRecorder(10),
-		egressQoSWg:  &sync.WaitGroup{},
-		egressSVCWg:  &sync.WaitGroup{},
-		anpWg:        &sync.WaitGroup{},
-
+		asf:                        asf,
+		fakeRecorder:               record.NewFakeRecorder(10),
+		egressQoSWg:                &sync.WaitGroup{},
+		egressSVCWg:                &sync.WaitGroup{},
+		anpWg:                      &sync.WaitGroup{},
+		fexec:                      fexec,
```

Comment:
Looks like every FakeOVN object uses the Start method, so why not have NewFakeOVN directly set:
```
fexec: ovntest.NewFakeExec(),
```
?

Reply:
you need to have access to fexec outside this function,
unless we export Fexec and then it is possible, but it means it will always export it even when not needed
so tried to do it as close as to current (i.e only when needed)

Reply:
yeah, I just figured everyone who initializes fakeOVN also used Start (also checked on 4 filed..) so if it was possible to always use that it would've made the code much simpler, even at the cost of making fexec public.
But if you say you checked it then it must be more complicated than I assumed so nevermind.

Reply:
adapted